orderapi/src/main/java/fi/hel/verkkokauppa/order/service/invoice/InvoiceService.java

```diff
@@ -15,14 +18,27 @@ public class InvoiceService {
 
     final InvoiceMapper invoiceMapper;
 
+    final IncrementId incrementId;
+
     @Autowired
-    public InvoiceService(OrderRepository orderRepository, InvoiceMapper invoiceMapper) {
+    public InvoiceService(OrderRepository orderRepository, InvoiceMapper invoiceMapper, IncrementId incrementId) {
         this.orderRepository = orderRepository;
         this.invoiceMapper = invoiceMapper;
+        this.incrementId = incrementId;
     }
 
-    public Order saveInvoiceToOrder(InvoiceDto invoice, Order order) {
-        order.setInvoice(invoiceMapper.fromDto(invoice));
+    public Order saveInvoiceToOrder(InvoiceDto invoiceDto, Order order) {
+        Invoice invoice = invoiceMapper.fromDto(invoiceDto);
+        invoice.setInvoiceId(order.getInvoice() == null || order.getInvoice().getInvoiceId().isEmpty()
```

Comment:
Could you wrap this to new function, to make it easier to test and mock. And I like to see this kind of multi conditions inside of if statement. Ternary is very good to use for.  like, setInvoiceIdFromOrder()